ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/ProjectNode.java

```diff
@@ -45,7 +46,12 @@ public ProjectNode(
         keyColumnNames,
         getSelectExpressions(),
         buildContext.buildNodeContext(getId().toString()),
-        buildContext
+        buildContext,
+        getFormatInfo()
     );
   }
+
+  public FormatInfo getFormatInfo() {
+    return getLeftmostSourceNode().getDataSource().getKsqlTopic().getValueFormat().getFormatInfo();
```

Comment:
Does this format hold for anything that the leftmost source may join with?

Reply:
@guozhangwang yes, it is sufficient